CMakeLists.txt

```diff
@@ -105,12 +105,19 @@ endif()
 #Install
 #------------------------------------------------------------------
 
-install(CODE "execute_process(COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/deps/gptp/linux/build/obj/daemon_cl ${CMAKE_INSTALL_PREFIX}/bin/daemon_cl)")
-install(CODE "execute_process(COMMAND mkdir -p /etc/alsa/conf.d)")
+install(PROGRAMS ${CMAKE_CURRENT_SOURCE_DIR}/deps/gptp/linux/build/obj/daemon_cl DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 
-install(CODE "execute_process(COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.tmp)")
-install(CODE "execute_process(COMMAND sed -i s@<lib_path>@${CMAKE_INSTALL_LIBDIR}@g ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.tmp)")
-install(CODE "execute_process(COMMAND mv ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.tmp /etc/alsa/conf.d/51-smartx.conf)")
+set(INSTALL_LIBDIR ${CMAKE_INSTALL_LIBDIR}/alsa-lib)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf.in ${CMAKE_CURRENT_SOURCE_DIR}/public/res/51-smartx.conf @ONLY)
+
+if(ASOUND_VERSION VERSION_LESS 1.1.7)
+        set(ALSA_CONF_PATH "/usr/share/alsa/alsa.conf.d")
+else()
+        install(DIRECTORY DESTINATION /etc/alsa/conf.d)
```

Comment:
Is it really necessary to create this dir?

Reply:
yes